Source/Charts/Charts/PieRadarChartViewBase.swift

```diff
@@ -677,13 +677,25 @@ open class PieRadarChartViewBase: ChartViewBase
         }()
 
         // Remove samples older than our sample time - 1 seconds
-        // while keeping at least one samples
-        let index = velocitySamples
-            .dropLast()
-            .lastIndex { $0.time < currentSample.time - 1 }
-        if let index = index {
-            velocitySamples.remove(at: index)
+        // while keeping at least one sample
+        
+        var i = 0, count = velocitySamples.count
```

Comment:
I'm not sure what I was doing above. 
```
let index = velocitySamples.firstIndex { currentSample.time - velocitySamples[i].time > 1.0 }
    ?? velocitySamples.endIndex - 1
velocitySamples = velocitySamples.suffix(from: index)
```

Reply:
So do you want to add another commit over this PR? Then we will merge it (please test the code first to see that all irrelevant samples are being dropped)

Reply:
@jjatie I'm merging this, please add a PR for a *Swiftier* version when you have the time :-)